Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -261,6 +261,42 @@ public final class KafkaProducer: Service, Sendable {
             return KafkaProducerMessageID(rawValue: newMessageID)
         }
     }
+    
+    func initTransactions(timeout: Duration = .seconds(5)) async throws {
```

Comment:
I think we definitely want to create a new `KafkaTransactionalProducer` type so that the APIs stay clear and separated between the normal producer and the transaction one.

Additionally, I think the APIs should look a bit different. I can imagine something like this

```
public func withTransaction<Result>(_ body: @Sendable (KafkaTransaction) async throws -> Void) async throws -> Result {
    // init transaction
    // begin transaction
    // call out to user code and pass a new `Transaction` struct to them
    // at the end call abort or finish transaction for them
}

```

Reply:
That makes perfect sense to have a separate producer for transactions. 
We have it at a higher level but that might be done here